src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -294,13 +294,41 @@ private BoundIndexerAccess TransformIndexerAccess(BoundIndexerAccess indexerAcce
             // tempy = Y()
             // tempc[tempx, tempy, 123] = tempc[tempx, tempy, 123] + 1;
 
-            ImmutableArray<BoundExpression> rewrittenArguments = VisitList(indexerAccess.Arguments);
-
             SyntaxNode syntax = indexerAccess.Syntax;
             PropertySymbol indexer = indexerAccess.Indexer;
+            ImmutableArray<int> argsToParamsOpt = indexerAccess.ArgsToParamsOpt;
             ImmutableArray<RefKind> argumentRefKinds = indexerAccess.ArgumentRefKindsOpt;
+
+            ImmutableArray<BoundExpression> rewrittenArguments = VisitArguments(
+                syntax,
+                indexerAccess.Arguments,
+                indexer,
+                argsToParamsOpt,
+                argumentRefKinds,
+                ref transformedReceiver!,
+                out ArrayBuilder<LocalSymbol>? argumentTemps,
+                out bool receiverIsSideEffectSequence);
```

Comment:
>out bool receiverIsSideEffectSequence [](http://example.com/codeflow?start=16&length=37)

It feels like the unwrapping would apply to any ```BoundSequence``` representing the ```transformedReceiver```. We wouldn't want to evaluate sideffects twice. Given that, I don't think we need this parameter. We should always deconstruct a sequence, Sideeffects go into stores, Locals go into ```temps```.